cpp/include/cuspatial/experimental/type_utils.hpp

```diff
@@ -41,6 +50,31 @@ struct vec_2d_to_tuple {
 
 }  // namespace detail
 
+/**
+ * @addtogroup type_factories
+ * @{
+ */
+
+/**
+ * @brief Create an iterator to `vec_2d` data from two input iterators.
+ *
+ * Interleaves x and y coordinates from separate iterators into a single iterator to x-y
+ * coordinates.
+ *
+ * @tparam VectorType cuSpatial vector type, must be `vec_2d`, `lonlat_2d` or `cartesian_2d`
+ * @tparam FirstIter Iterator type to the first component of `vec_2d`. Must meet the requirements of
+ * [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam SecondIter Iterator type to the second component of `vec_2d`. Must meet the requirements
+ * of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @param first Iterator to beginning of `vec_2d::x`
+ * @param second Iterator to beginning of `vec_2d::y`
+ * @return Iterator to `vec_2d`
+ *
+ * @pre `first` and `second` must iterate on same data type.
```

Comment:
Maybe out of scope for this PR, but if first and second must iterate on the same type why have two template parameters `FirstIter` and `SecondIter`? Am I misunderstanding this condition?

Reply:
They must iterator on the same value type, but the iterator types may differ.

Consider these two values which could be iterators with the same value type:
```
float* values_a;

auto values_b = thrust::make_transform_iterator(
  thrust::make_counting_iterator(0),
  [](auto idx){ return static_cast<float>(idx);
});

Reply:
Correct. An STL example is the two-input flavor of `transform`. This enables the use of custom or fancy iterators, as well as raw pointers, etc.

Reply:
Ah of course I didn't realize this was referring to the underlying data type, not the iterator type.